### ISSUE TITLE
Storyshot fails to respect threshold settings

### ISSUE THREAD

**Issue by patthiel** [storybooks/storybook#3801]
If you are reporting a bug or requesting support, start here:
### Bug or support request summary

I've setup storyshots in this [PR](https://github.com/linode/manager/pull/3475) for use in the Linode Manager project. Running the tests locally passes just fine, but in Travis one of the tests consistently fails the comparison (ranging between a 4 - 6% difference).. I tried compensating for this by bumping the failure threshold to 10 percent, then to 20 percent. Nothing seems to be doing the trick.  The travis results can be found here: https://api.travis-ci.org/v3/job/396935367/log.txt

```
Expected image to match or be a close match to snapshot
but was 4.000048562548562% different from snapshot (82369 differing pixels).

See diff for details:
/home/travis/build/linode/manager/src/components/__image_snapshots__/__diff_output__/storyshots-test-tsx-image-storyshots-selection-row-interactive-example-1-diff.png
```

Is there something wrong with my configuration? https://github.com/patthiel/manager/blob/1c4a9195d62fc9c489ca7a060d4b275e53cc4f24/src/components/Storyshots.test.tsx

```javascript
import initStoryshots, { imageSnapshot } from '@storybook/addon-storyshots';

const getMatchOptions = ({context : {kind, story}, url}) => ({
    // Allow 20% difference
    failureThreshold: 0.2,
    failureThresholdType: 'percent',
    customDiffConfig:  { threshold: 0.15 }
});

jest.setTimeout(30000);

initStoryshots({
    suite: 'Image storyshots',
    test: imageSnapshot({storybookUrl: 'http://localhost:6006', getMatchOptions})
});
```

### Steps to reproduce

* See linked PR and Travis build.

### Please specify which version of Storybook and optionally any affected addons that you're running:

@storybook/react@^3.4.7
@storybook/addon-storyshots@3.4.8

**Comment by patthiel** [comment]
@igor-dv @shilman any ideas here? We'd really like to see this integrated into the Linode [Manager](https://github.com/linode/manager/) test suite.

**Comment by igor-dv** [comment]
Maybe @thomasbertet can suggest something

**Comment by igor-dv** [comment]
BTW, dod you try to use `beforeScreenshot` like [here](https://github.com/storybooks/storybook/tree/master/addons/storyshots/storyshots-puppeteer#specifying-options-to-jest-image-snapshots) ?

**Comment by patthiel** [comment]
@igor-dv  I just tried adding that, didn't seem to help.. here's the [config](https://github.com/patthiel/manager/blob/698bf9e34d75f5b4eb064430736ec0a5a4921dc2/src/components/Storyshots.test.tsx)

And the build failure [here](https://travis-ci.org/linode/manager/builds/403222505?utm_source=github_status&utm_medium=notification) - Same issue.

**Comment by igor-dv** [comment]
Yeah, it's not so clear. Did you see those problematic generated images somehow? Like what is the difference?

**Comment by patthiel** [comment]
@igor-dv here's the diff image: 
![image](https://user-images.githubusercontent.com/12736578/42767734-ca718508-88eb-11e8-9e19-0c21d0c66111.png)

Seems like manually settting the [viewport](https://github.com/GoogleChrome/puppeteer/blob/master/docs/api.md#pagesetviewportviewport) for the tests could address the issue, although, i'm not sure where in the config i can do that?

**Comment by igor-dv** [comment]
I see the place where it starts breaking

![image](https://user-images.githubusercontent.com/7867954/42769467-dec5a568-892a-11e8-9979-05a81c4034d5.png)

In the latest alpha, you have the ability to [define](https://github.com/storybooks/storybook/tree/master/addons/viewport) viewports for stories, I am not sure this is the problem though. 

Does your ci run under the same system preferences as your local env? Most of the problems are related to the font differences between a local env and the CI machine.

**Comment by patthiel** [comment]
@igor-dv  ah, ok. Interesting..

So, I managed to get it working by updating and running everything via docker-compose. This will ensure system preferences are always the same. (That process was not as straightforward as i had hoped, but i found the puppeteer [workaround](https://github.com/storybooks/storybook/blob/master/scripts/workaround-puppeteer-issue-290.sh)  helped resolve the issues i ran into.)

Unfortunately, it doesn't seem like we solved the underlying issue initially reported regarding threshold settings not being respected, but i am at least happy to have  integrated storyshot into our project. When I have time, I will attempt to dig more into this issue.

For those that may find this thread, [here](https://github.com/linode/manager/pull/3552) is my PR using docker-compose to run storybook/storyshots. a few improvements will be made in the next pass.
